Library/Homebrew/exceptions.rb

```diff
@@ -65,6 +65,25 @@ def to_s; <<-EOS.undent
   end
 end
 
+class TapFormulaAmbiguityError < RuntimeError
+  attr_reader :name, :paths, :formulae
+
+  def initialize name, paths
+    @name = name
+    @paths = paths
+    @formulae = paths.map do |path|
+      path.to_s =~ HOMEBREW_TAP_PATH_REGEX
+      "#{$1}/#{$2.sub("homebrew-", "")}/#{path.basename(".rb")}"
+    end
+
+    super <<-EOS.undent
+      Multiple formulae found: #{formulae.map { |f| "\n       * #{f}" }.join}
```

Comment:
Yes, current implementation will prioritize core formula. But it can be easily changed if wanted.

>  I think we want to be able to use Homebrew without core formulae at all and use only a tap.

I am always confused about how we `depends_on` another formula in the same tap for a tap formula. Should we use full qualified name or not? Deep down is that we need to solve #14089 first.